src/core/reactor.cc

```diff
@@ -825,15 +835,15 @@ class backtrace_buffer {
             append("0x");
             append_hex(f.addr);
             append("\n");
-        });
+        }, _immediate);
     }
 
     void append_backtrace_oneline() noexcept {
         backtrace([this] (frame f) noexcept {
             reserve(3 + sizeof(f.addr) * 2);
             append(" 0x");
             append_hex(f.addr);
-        });
+        }, _immediate);
```

Comment:
Thinking out loud. This transparently converts "immediate" to "incremental". Oh, well

Reply:
> Instead, introduce an "immediate" mode to backtrace buffer which prints all appended strings immediately

Hopefully it won't intermix these lines with more logs (from other shards?) thus making the whole thing unreadable. On the other hand, SIGSEGV shouldn't be easy to debug, and having intermixed lines is still better than having nothing at all

Reply:
> Thinking out loud. This transparently converts "immediate" to "incremental". Oh, well

Yeah. I felt that these were the right names for the two concepts, as they are not the same and could in principle be set independently, but here in the only existing usage "immediate" directly implies "incremetal". I could imagine a future where incremental would be a separate option inside the backtrace buffer in case you wanted say "non-immediate, incremental" or whatever.

Reply:
> Hopefully it won't intermix these lines with more logs (from other shards?) thus making the whole thing unreadable.

Yeah. On the one hand it's possible, on the other hand it will hopefully be rare as we do make all these calls back-to-back with no delay in between so the window to race with concurrent output on another shard is small so I expect tearing to be rare in practice, and in on the third hand:

> On the other hand, SIGSEGV shouldn't be easy to debug, and having intermixed lines is still better than having nothing at all

Yup, exactly.

Reply:
BTW, this is one reason to do this only on the segfault path, not on the other paths (abort handler, reactor stall notifier, etc).